File: tests/fixtures/godemo/storage/repository.go
package storage

// User is a user record.
type User struct {
	ID   uint64
	Name string
}

// Repository is the storage boundary the domain layer consumes.
type Repository interface {
	// FindUser finds a single user by id.
	FindUser(id uint64) (*User, error)
	// ListUsers lists every user.
	ListUsers() ([]User, error)
}

File: tests/fixtures/scalademo/src/main/scala/storage/Repository.scala
package storage

/** A user record. */
case class User(id: Long, name: String)

/** The storage boundary the domain layer consumes. */
trait Repository:
  /** Find a single user by id. */
  def findUser(id: Long): Option[User]

  /** List every user. */
  def listUsers(): List[User]
